datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -230,6 +254,32 @@ public void initializeDatastream(Datastream stream, List<Datastream> allDatastre
     coordinator.stop();
   }
 
+  @Test
+  public void testConnectorGracefullyShutdown() throws Exception {
+    String testCluster = "testConnectorGracefullyShutdown";
+    String testConnectorType = "testConnectorType";
+    AtomicBoolean eventProcessed = new AtomicBoolean(false);
+    Coordinator instance1 = createCoordinator(_zkConnectionString, testCluster, new Properties(),
+        new DummyTransportProviderAdminFactory(), (event) -> {
+          try {
+            Thread.sleep(5000);
```

Comment:
1. I understand you're trying to accomplish but please add a comment explaining the purpose of this delay.
2. Please, consider reducing this timeout; adding a 5-second delay to every build for a single test sounds a bit much.